corgi/tasks/brew.py

```diff
@@ -168,6 +130,70 @@ def slow_fetch_brew_build(
     return build_created or root_created or any_child_created
 
 
+def _get_nested_builds(
+    build_type: str, nested_builds: set[int], force_process: bool, save_product: bool
+) -> None:
+    logger.info("Fetching brew builds for (%s, %s)", nested_builds, build_type)
+    for b_id in nested_builds:
+        logger.info("Requesting fetch of nested build: (%s, %s)", b_id, build_type)
+        slow_fetch_brew_build.delay(
+            b_id, build_type, save_product=save_product, force_process=force_process
+        )
+
+
+def _check_and_load_errata(released_errata_tags, force_process) -> None:
+    """ " for builds with released errata tags set ProductComponentRelation"""
+    for e in released_errata_tags:
+        slow_load_errata.delay(e, force_process=force_process)
+
+
+def _check_and_load_brew_tags(
+    build_id: str, tags: list[str], build_type: str, softwarebuild: SoftwareBuild
+) -> int:
+    # for builds with any tag, check if the tag is used for product stream relations, and create the
+    # relations if so.
+    if not tags:
+        logger.info("no brew tags")
+    else:
+        new_relations = load_brew_tags(softwarebuild, tags)
+        logger.info(f"Created {new_relations} for brew tags in {build_type}:{build_id}")
+
+
+def _save_children(child_components: list[dict], root_node: ComponentNode) -> bool:
+    any_child_created = False
+    for child_component in child_components:
+        any_child_created |= save_component(child_component, root_node)
+    return any_child_created
+
+
+def _get_completion_time(build_id: str, completion_time):
+    if completion_time:
+        dt = dateparse.parse_datetime(completion_time.split(".")[0])
+        if dt:
+            completion_dt = make_aware(dt)
+        else:
+            raise ValueError(f"Could not parse completion_time for build {build_id}")
+    else:
+        # Build has no timestamp even though it's in COMPLETE state?
+        # This really shouldn't happen
+        raise ValueError(f"No completion_time for build {build_id}")
+    return completion_dt
+
+
+def _check_and_save_type(
+    component: dict[str, Any], softwarebuild: SoftwareBuild, save_product: bool, build_id: str
+) -> Optional[Tuple[ComponentNode, bool]]:
```

Comment:
```suggestion
) -> tuple[Optional[ComponentNode], bool]:
```
It might be a little cleaner to always return a tuple below:
```
    else:
        logger.warning(f"Build {build_id} type is not supported: {component['type']}")
        return (None, False)
```
Then above can always unpack the function's return value directly:
```
    root_node, root_created = _check_and_save_type(component, softwarebuild, save_product, build_id)
    if root_node is None:
        return False
```